docs/reference/settings.md

```diff
@@ -278,6 +278,38 @@ bypasses SSL verification and could expose you to MITM attacks.
 
 ---
 
+### [`allow-insecure-host`](#allow-insecure-host) {: #allow-insecure-host }
+
+Allow insecure connections to host.
+
+Expects to receive either a hostname (e.g., `localhost`), a host-port pair (e.g.,
+`localhost:8080`), or a URL (e.g., `https://localhost`).
+
+WARNING: Hosts included in this list will not be verified against the system's certificate
+store. Only use `--allow-insecure-host` in a secure network with verified sources, as it
+bypasses SSL verification and could expose you to MITM attacks.
+
+**Default value**: `[]`
+
+**Type**: `list[str]`
+
+**Example usage**:
+
+=== "pyproject.toml"
+
+    ```toml
+    [tool.uv]
+    allow-insecure-host = ["localhost:8080"]
+    ```
+=== "uv.toml"
+
+    ```toml
+    
+    allow-insecure-host = ["localhost:8080"]
+    ```
+
+---
+
```

Comment:
Does this now appear multiple times? I see it's already included twice on main...

Reply:
I'm not clear about how the options flow, at least uvx on windows has some duplication (https://github.com/astral-sh/uv/actions/runs/11482045623/job/31954176059?pr=8476), i appreciate guidance on where allow insecure host is supposed to live (or not).

My basic thinking is:
* It needs to be in the global scope
* For backwards compatibility, it also needs to be in the pip scope, and take precedence over the global option.
The latter point seems to thread this option through a lot of other places.

Reply:
Ok, I can take a look at how that's popping up here.